src/main/java/liquibase/ext/spanner/snapshotgenerator/SequenceSnapshotGeneratorSpanner.java

```diff
@@ -0,0 +1,74 @@
+package liquibase.ext.spanner.snapshotgenerator;
+
+import liquibase.database.Database;
+import liquibase.ext.spanner.ICloudSpanner;
+import liquibase.snapshot.SnapshotGenerator;
+import liquibase.snapshot.jvm.SequenceSnapshotGenerator;
+import liquibase.statement.SqlStatement;
+import liquibase.statement.core.RawParameterizedSqlStatement;
+import liquibase.structure.DatabaseObject;
+import liquibase.structure.core.ForeignKey;
+import liquibase.structure.core.Schema;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SequenceSnapshotGeneratorSpanner extends SequenceSnapshotGenerator {
+
+  /**
+   * This generator will be in all chains relating to CloudSpanner, whether or not
+   * the objectType is {@link liquibase.structure.core.Sequence}.
+   */
+  @Override
+  public int getPriority(Class<? extends DatabaseObject> objectType, Database database) {
+    if (database instanceof ICloudSpanner) {
+      return PRIORITY_DATABASE;
+    }
+    return PRIORITY_NONE;
+  }
+
+  @Override
+  protected SqlStatement getSelectSequenceStatement(Schema schema, Database database) {
+    if (database instanceof ICloudSpanner) {
+      List<String> parameter = new ArrayList<>(2);
+      parameter.add(database.getDefaultCatalogName());
+      parameter.add(database.getDefaultSchemaName());
```

Comment:
Shouldn't this rather use the `schema` and `database` values that are given as arguments to this method?

Reply:
fixed